nav2_mppi_controller/src/optimizer.cpp

```diff
@@ -347,14 +360,22 @@ void Optimizer::updateStateVelocities(
   propagateStateVelocitiesFromInitials(state);
 }
 
-void Optimizer::updateInitialStateVelocities(
-  models::State & state) const
+void Optimizer::updateInitialStateVelocities(models::State & state) const
 {
-  state.vx.col(0) = static_cast<float>(state.speed.linear.x);
-  state.wz.col(0) = static_cast<float>(state.speed.angular.z);
+  const bool open = settings_.open_loop;
```

Comment:
Just use `settings_.open_loop` rather than `open` in the setting of vx/wz/vy. You have the line extension either way.